src/System.IO.Abstractions/FileStream.cs

```diff
@@ -0,0 +1,58 @@
+﻿namespace System.IO.Abstractions
+{
+    /// <summary>
+    /// A <see cref="Stream"/> that has properties similar to <see cref="System.IO.FileStream"/>
+    /// </summary>
+    public class FileStream : Stream
```

Comment:
I think we should use another name for this to prevent conflicts with `System.IO.FileStream`.

I don't have a good idea, maybe `FileStreamBase` to mirror other types here?